webapp/src/pages/createForm/components/ApplicationForm.js

```diff
@@ -0,0 +1,181 @@
+import React, { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next'
+import { eventService } from '../../../services/events';
+import icon from '../icon.svg';
+import Section from './Section';
+
+const ApplicationForm = (props) => {
+  const [nominate, setNominate] = useState(false);
+  const [event, setEvent] = useState({
+    loading: true,
+    event: null,
+    error: null,
+  });
+  const [sections, setSections] = useState([{
+    id: 1,
+    name: 'Untitled Section',
+    description: '',
+    questions: [
+      {
+        id: `${Math.random()}`,
+        order: 1,
+        headline: '',
+        placeholder: '',
+        type: null,
+        options: [],
+        value: '',
+        label: '',
+        required: false
+      }
+    ]
+  }]);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    eventService.getEvent(props.event.id).then( res => {
+      setEvent({
+        loading: res.loading,
+        event: res.event,
+        error: res.error
+      })
+    })
+  }, []);
+  const handleCheckChanged = (e) => {
+    setNominate(e.target.checked);
+  }
+  const handleSection = (input) => {
+    setSections(input)
+  }
+  const addSection = () => {
+    const id = sections.length + 1;
```

Comment:
> This is fine for creating a form from a blank slate, but will cause issues when editing an existing form where the IDs for existing sections come from the backend - in that case this code might cause an ID collision. The collision could be solved easily by using the maximum ID + 1, however that still leaves another problem, which is that the backend for the PUT request relies on the IDs to know whether to update or insert a section (update if ID is populated, insert if not). I know you rely on this ID later for adding questions, so either you'll have to add a surrogate ID property to the sections to use just in the front-end, or I don't know if there's a way to push the question state entirely into the Section component and recover it easily in the ApplicationForm component when you need it? (there isn't as far as I understand, but I don't know if the new Hooks stuff adds anything like this?)

Noted, Thanks for the heads up!
Yes these IDs will be used only for frontend purpose. In case of editing a existing form I will be handling this issue by renaming some properties or using aliases.